src/Passes/docs/src/DeveloperGuide/QuickStart.md

```diff
@@ -0,0 +1,150 @@
+
+```
+namespace TeleportChain {
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Canon;
+    open Microsoft.Quantum.Arrays;
+    open Microsoft.Quantum.Measurement;
+    open Microsoft.Quantum.Preparation;
+
+    operation PrepareEntangledPair(left : Qubit, right : Qubit) : Unit is Adj + Ctl {
+        H(left);
+        CNOT(left, right);
+    }
+
+    operation ApplyCorrection(src : Qubit, intermediary : Qubit, dest : Qubit) : Unit {
+        if (MResetZ(src) == One) { Z(dest); }
+        if (MResetZ(intermediary) == One) { X(dest); }
+    }
+
+    operation TeleportQubitUsingPresharedEntanglement(src : Qubit, intermediary : Qubit, dest : Qubit) : Unit {
+        Adjoint PrepareEntangledPair(src, intermediary);
+        ApplyCorrection(src, intermediary, dest);
+    }
+
+    operation TeleportQubit(src : Qubit, dest : Qubit) : Unit {
+        use intermediary = Qubit();
+        PrepareEntangledPair(intermediary, dest);
+        TeleportQubitUsingPresharedEntanglement(src, intermediary, dest);
+    }
+
+    operation DemonstrateEntanglementSwapping() : (Result, Result) {
+        use (reference, src, intermediary, dest) = (Qubit(), Qubit(), Qubit(), Qubit());
+        PrepareEntangledPair(reference, src);
+        TeleportQubit(src, dest);
+        return (MResetZ(reference), MResetZ(dest));
+    }
+
+    @EntryPoint()
+    operation DemonstrateTeleportationUsingPresharedEntanglement() : Unit {
+        let nPairs = 2;
+        use (leftMessage, rightMessage, leftPreshared, rightPreshared) = (Qubit(), Qubit(), Qubit[nPairs], Qubit[nPairs]);
+        PrepareEntangledPair(leftMessage, rightMessage);
+        for i in 0..nPairs-1 {
+            PrepareEntangledPair(leftPreshared[i], rightPreshared[i]);
+        }
+
+        TeleportQubitUsingPresharedEntanglement(rightMessage, leftPreshared[0], rightPreshared[0]);
+        for i in 1..nPairs-1 {
+            TeleportQubitUsingPresharedEntanglement(rightPreshared[i-1], leftPreshared[i], rightPreshared[i]);
+        }
+
+        let _ = MResetZ(leftMessage);
+        let _ =  MResetZ(rightPreshared[nPairs-1]);
+    }
+}
+```
+
+The corresponding QIR can be generated by going to `examples/QubitAllocationAnalysis` and run `make analysis-example.ll`. This step requires that you have a working installation of Q#. Once compiled and the initial QIR is generated and saved in the file `analysis-example.ll`, we execute the command
+
+```
+./Source/Apps/qat --generate --profile baseProfile -S ./analysis-example.ll
+```
+
+The QAT tool will now attempt to map the QIR in `analysis-example.ll` into a QIR which is compatible with the base format. Removing type and function declarations, the correspoding code reads:
+
+```
+; ModuleID = './analysis-example.ll'
+source_filename = "./analysis-example.ll"
+
+define internal fastcc void @TeleportChain__DemonstrateTeleportationUsingPresharedEntanglement__body() unnamed_addr {
+entry:
+  call void @__quantum__qis__h(%Qubit* null)
```

Comment:
Just FYI, with the coming update to the newer beta package, these now show up with `__body` at the end of the names. See the changes to the old readme here: https://github.com/microsoft/qsharp-compiler/pull/1121.
It probably makes sense to merge this PR first, since it includes build fixes, and then update 1121 to include changes to these files.

Reply:
Thanks for the heads up. Do we have a full list of all symbol names somewhere? If so, I might just scan the entire Passes directory for invalid names.

Reply:
We don't have that full list in one place, no, but it comes from the change made to the runtime repo. In #1121 I went ahead and did what you suggest: searched the whole Passes directory and double checked every use of a `__quantum__qis__` function to make sure I either regenerated the corresponding file or manually updated the references. I can just do that again after this PR merges.

Reply:
NIce, thanks!